### ISSUE TITLE
manual trigger: hold list not used

### ISSUE THREAD

**Issue by hjoliver** [cylc/cylc-flow#6513]
Future tasks can be held - they're recorded in a "hold list" so that once spawned into the active pool they will be held.
This works as advertised for natural triggering, but not for manual triggering.

Current behaviour for manually triggering a held task:
 1. existing held tasks (e.g. waiting with partially satisfied prerequisites):
    - task runs but retains the held attribute
 2. future held task:
    -  task runs without having the held attribute applied 

I think 1. is correct behaviour, and 2. is a bug -it should run as held as for 1.

ASIDE: I suppose we *could* decree that held tasks will not trigger naturally or manually, but I think it's more reasonable to have them run (I would not do a manual trigger if I didn't want it to run) but retain the held state to prevent further job submission.

**Comment by oliver-sanders** [comment]
> I think 1. is correct behaviour, and 2. is a bug - it should run as held (same as 1.)

Agreed (remove question label if happy with this).

> I suppose we could decree that held tasks will not trigger naturally or manually, but I think it's more reasonable to have them run 

Manual trigger should always override the held state.

**Comment by oliver-sanders** [comment]
I think this has been fixed by recent changes.

Here's my example:

```cylc
[scheduling]
    cycling mode = integer
    initial cycle point = 1
    runahead limit = P1
    [[graph]]
        R/1 = a
        R/2 = a[^]:started & a[^] => b  # a:started ensures 2/b is n=0
        R/3 = b[-P1] => x
        R/4 = b

[runtime]
    [[a]]
        script = """
            sleep 5

            cylc hold "$CYLC_WORKFLOW_ID//" //2/b //4/b

            # trigger n=0 task
            cylc trigger "$CYLC_WORKFLOW_ID//2/b"

            # trigger n=1 task
            cylc trigger "$CYLC_WORKFLOW_ID//4/b"
        """
    [[b]]
        script = false
        execution retry delays = PT0S
    [[x]]
```

This triggers 2/b (n=0) and 4/b (n=1).

Both tasks retain the held modifier and neither are automatically retried as configured:

```
      - ● 1
         + ● ■ a
      - ̎○ 2
         - ̎○ ■ b
            + #01 ■
      - ○ 3
           ○ x
      - ̎○ 4
         - ̎○ ■ b
            + #01 ■
```

@hjoliver, happy to close this as fixed?

**Comment by dwsutherland** [comment]
> So we just must a datastore update call?

I think he's pointing out that the task should not be released as part of the group trigger? (i.e. should remain in held state)

**Comment by oliver-sanders** [comment]
> I think he's pointing out that the task should not be released as part of the group trigger? (i.e. should remain in held state)

Yes, if that is what we want, we are doing the opposite! May need to recap the proposal to work out why this release is there.
